PrimeSwift/solution_1/Sources/PrimeSieveSwift/main.swift

```diff
@@ -92,40 +70,61 @@ extension PrimeSieveSwift {
     }
 }
 
-extension PrimeSieveSwift {
+extension Sieve {
     func printResults(showingNumbers: Bool, duration: TimeInterval, passes: Int) {
         if showingNumbers {
             print("2, ", terminator: "")
-        }
-
-        var count = 1
-        for num in stride(from: 3, to: sieveSize, by: 2) {
-            if bits.getBit(num) {
-                if showingNumbers {
+            for num in stride(from: 3, to: sieveSize, by: 2) {
+                if bits[index(num)] {
                     print(num, terminator: ", ")
                 }
-                count += 1
             }
         }
 
-        assert(count == countPrimes())
-
-        print("\nPasses: \(passes), Time: \(duration), Avg: \(duration/Double(passes)), Limit: \(sieveSize), Count: \(count), Valid: \(validateResults())")
+        print("\nPasses: \(passes), Time: \(duration), Avg: \(duration/Double(passes)), Limit: \(sieveSize), Count: \(countPrimes()), Valid: \(validateResults())")
         
         /// Following 2 lines added by rbergen to conform to drag race output format
         print()
-        print("j-f1;\(passes);\(duration);1;algorithm=base,faithful=yes\n")
+        print("yellowcub;\(passes);\(duration);1;algorithm=base,faithful=yes,bits=\(8*MemoryLayout<Bool>.size)\n")
     }
 }
 
-let start = Date()
-var passes = 0
-var sieve: PrimeSieveSwift!
+struct StopWatch {
+    private let start: Date
```

Comment:
nit: you can initialize `start` here so that the `init()` method does not need to be explicitly specified:

```suggestion
    private let start = Date()
```

Reply:
Agreed, that looks cleaner.